southpark-search/tests/test_southparksearch.py

```diff
@@ -1,44 +1,36 @@
-__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
-__license__ = "Apache-2.0"
-
-import pytest
-
 import json
 import os
 import sys
 import shutil
 import subprocess
 
 from jina.flow import Flow
+import pytest
 
-os.chdir(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
 
 num_docs = 100
 top_k = 3
-index_flow_file_path = "flow-index.yml"
-query_flow_file_path = "flow-query.yml"
-
+index_flow_file_path = 'flow-index.yml'
+query_flow_file_path = 'flow-query.yml'
 
-def config():
-    os.environ["JINA_DATA_FILE"] = os.environ.get("JINA_DATA_FILE", "tests/data-index.csv")
-    os.environ["JINA_WORKSPACE"] = os.environ.get("JINA_WORKSPACE", "workspace")
-    os.environ["JINA_PORT"] = os.environ.get("JINA_PORT", str(45678))
 
-
-def prepare_data():
-    pass
+def config(tmpdir):
+    os.environ['JINA_DATA_FILE'] = 'tests/data-index.csv'
+    os.environ['JINA_WORKSPACE'] = str(tmpdir)
+    os.environ['JINA_PORT'] = str(45678)
 
 
+@pytest.fixture
 def setup_env():
-    subprocess.check_call([sys.executable, "-m", "pip", "install", "-r", "requirements.txt"])
-    subprocess.check_call([sys.executable, "-m", "pip", "install", "requests", "jina[http]"])
-
+    os.chdir(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
```

Comment:
should it be changed back afterwards'

Reply:
All tests will require to use the pods and will require cwd as the example's root and will have to run this chdir to be there. So I don't think we need to change it back to examples root although no harm in doing it.

Reply:
I don't like this. Using `os.chdir` always is a code/setup smell for me. Why not prefix all needed files with `southpark-search`, if the tests are always run from examples main directory? With this, you don't need to use such a magic.